### PR TITLE
Get trust

### PR DESCRIPTION
**Pull request by versatil56**

### CODE REVIEW

File: app/uk/gov/hmrc/trustregistration/models/EmploymentTrust.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                           beneficiaries: Beneficiaries,
+                           isEmployerFinancedRetirementBenefitScheme: Boolean,
+                           employerFinancedRetirementBenefitSchemeStartDate: Option[DateTime] = None) {
+
+  private val atleastOneTypeOfAsset: Boolean = ((assets.monetaryAssets.isDefined && assets.monetaryAssets.get.size > 0) ||
+    (assets.propertyAssets.isDefined && assets.propertyAssets.get.size > 0) ||
+    (assets.shareAssets.isDefined && assets.shareAssets.get.size > 0) ||
+    (assets.businessAssets.isDefined && assets.businessAssets.get.size > 0))
+
+  private val atleastOneRequiredBenefciary: Boolean = (beneficiaries.individualBeneficiaries.isDefined ||
```
Comment (chriswaugh) [inline app/uk/gov/hmrc/trustregistration/models/EmploymentTrust.scala]: Is it worth also checking i.e. individualBeneficiaries.size > 0 as well to check that both the lists are defined and that one of them has at least one item in it

File: app/uk/gov/hmrc/trustregistration/models/WillIntestacyTrust.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+case class WillIntestacyTrust(assets: Assets, beneficiaries: Beneficiaries, deceased: Individual) {
+
+  private val atleastOneTypeOfAsset: Boolean = ((assets.monetaryAssets.isDefined && assets.monetaryAssets.get.size > 0) ||
+    (assets.propertyAssets.isDefined && assets.propertyAssets.get.size > 0) ||
+    (assets.shareAssets.isDefined && assets.shareAssets.get.size > 0)||
+    (assets.businessAssets.isDefined && assets.shareAssets.get.size > 0) ||
+    (assets.otherAssets.isDefined && assets.otherAssets.get.size > 0))
+
+
+  private val containsRequiredBeneficiaries: Boolean = (beneficiaries.individualBeneficiaries.isDefined ||
```
Comment (chriswaugh) [inline app/uk/gov/hmrc/trustregistration/models/WillIntestacyTrust.scala]: size check here too?

File: test/resources/ValidBusinessAsset.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "buildingName" : "Test",
+  "payeRef" : "Test",
+  "description" : "This is a description",
+  "correspondenceAddress" : "{ADDRESS}",
+  "value" : 1234,
+  "lastValudationDate" : "1940-04-04"
```
Comment (chriswaugh) [inline test/resources/ValidBusinessAsset.json]: typo on lastValidationDate

Comment (chriswaugh) [inline test/resources/ValidBusinessAsset.json]: or is it lastValuationDate?

File: test/resources/InvalidWillInstestacyTrust.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (chriswaugh) [inline test/resources/InvalidWillInstestacyTrust.json]: This file isn't referenced anywhere, can it be removed?

File: app/uk/gov/hmrc/trustregistration/models/EmploymentTrust.scala (hunk 1 above)
Comment (versatil56) [inline app/uk/gov/hmrc/trustregistration/models/EmploymentTrust.scala]: This check is already being done in the parent class.

File: app/uk/gov/hmrc/trustregistration/models/WillIntestacyTrust.scala (hunk 2 above)
Comment (versatil56) [inline app/uk/gov/hmrc/trustregistration/models/WillIntestacyTrust.scala]: Just like the other, size checked in parent class.

File: test/resources/InvalidWillInstestacyTrust.json (hunk 4 above)
Comment (versatil56) [inline test/resources/InvalidWillInstestacyTrust.json]: Thanks for this, this file should have been used by object "invalidWillIntestacyTrustJson" in DataExamples. It is now being used.

**Pull request merged**